src/blas/backends/cublas/cublas_batch.cpp

```diff
@@ -167,12 +167,25 @@ inline void gemm_batch_impl(sycl::queue &queue, transpose transa, transpose tran
             auto b_ = sc.get_mem<cuTypeB *>(b_acc);
             auto c_ = sc.get_mem<cuTypeC *>(c_acc);
             cublasStatus_t err;
-            CUBLAS_ERROR_FUNC_T_SYNC(
-                "cublasGemmStridedBatchedEx", cublasGemmStridedBatchedEx, err, handle,
-                get_cublas_operation(transa), get_cublas_operation(transb), m, n, k, &alpha, a_,
-                get_cublas_datatype<cuTypeA>(), lda, stride_a, b_, get_cublas_datatype<cuTypeB>(),
-                ldb, stride_b, &beta, c_, get_cublas_datatype<cuTypeC>(), ldc, stride_c, batch_size,
-                get_cublas_datatype<cuTypeS>(), cublas_gemm_algo);
+#ifdef SYCL_EXT_ONEAPI_ENQUEUE_NATIVE_COMMAND
+            CUBLAS_ERROR_FUNC_T("cublasGemmStridedBatchedEx", cublasGemmStridedBatchedEx,
+                                err, handle, get_cublas_operation(transa),
+                                get_cublas_operation(transb), m, n, k, &alpha, a_,
+                                get_cublas_datatype<cuTypeA>(), lda, stride_a, b_,
+                                get_cublas_datatype<cuTypeB>(), ldb, stride_b, &beta, c_,
+                                get_cublas_datatype<cuTypeC>(), ldc, stride_c, batch_size,
+                                get_cublas_datatype<cuTypeS>(), cublas_gemm_algo);
+#else
+            CUBLAS_ERROR_FUNC_T_SYNC("cublasGemmStridedBatchedEx",
+                                     cublasGemmStridedBatchedEx, err, handle,
+                                     get_cublas_operation(transa),
+                                     get_cublas_operation(transb), m, n, k, &alpha, a_,
+                                     get_cublas_datatype<cuTypeA>(), lda, stride_a, b_,
+                                     get_cublas_datatype<cuTypeB>(), ldb, stride_b, &beta,
+                                     c_, get_cublas_datatype<cuTypeC>(), ldc, stride_c,
+                                     batch_size, get_cublas_datatype<cuTypeS>(),
+                                     cublas_gemm_algo);
+#endif
```

Comment:
Can't we use `cublas_native_named_func` in this file?

Reply:
Yeah so for the few cases that I didn't use `cublas_native_named_func` this is because the functions don't pass the function name as a template parameter, so that I cannot use the generic function `cublas_native_named_func`.
I thought about having a neater solution for those cases also, but in the end I decided to just do it like this because there aren't many cases like this, and also I'm not sure if there is a clean way to do it without making quite a big change to the implementation. 
Although I didn't think too hard about this, there might be a better way.

Reply:
Ah maybe I can..

Reply:
It seems that for some reason the compiler cannot infer the type of the cublas functions ending in `*Ex`. I don't know what that is, but seems that I cannot use the `cublas_native_named_func` in those cases. However I spotted that there was one case in this file where I should have used it and didn't, so I've just updated that.

Reply:
That's unfortunate, thanks for checking.